Communikate CK20Page refactor and clean

zelmon64 opened the pull request:
Refactor and possible bug fixes in the CK20Page class(es) as outlined in #375.

This mainly targets `CKPageFileChanged` but also improves `dec2hex` and the background colour setters.

kmcnaught commented on src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs:
```diff
@@ -1130,7 +622,7 @@ public string CKText_10
+            set { SetValue(CKBaCo_10Property, value); }
         }
 
         public static readonly DependencyProperty CKImSo_10Property =
```
For each of the CKImSo_xy setters (a few lines below, I can't comment there) could you please move the logic into one place outside of the setter, just like you did for rgb2hex. This makes it easier to modify if the logic changes later.

zelmon64 replied:
Thanks for reminding me about this. The logic is now done on lines [168-170](https://github.com/zelmon64/OptiKey/blob/29b1d2ace719f6e1f8a2f0b50537a164227aab2b/src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs#L168-L170).

kmcnaught commented:
Hi @zelmon64 , could you please also remove the unused CKKey files from the project in this PR?

zelmon64 commented on src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs:
```diff
@@ -68,46 +68,48 @@ private static void CKPageFileChanged(DependencyObject dependencyObject, Depende
+                    blankbutton.load_board = defaultBoard;
 
                     for (int r = 1; r < 5; ++r)
+                    {
```
I've added some comments but also rearranged it to a simpler method. If the comments aren't enough, please say so and I'll add more.

zelmon64 commented on src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs:
```diff
@@ -68,46 +68,48 @@ private static void CKPageFileChanged(DependencyObject dependencyObject, Depende
-                                if (CKPageOBF.buttons.ElementAt(b).id != c.ToString() + r.ToString())
+                                if (CKPageOBF.buttons.ElementAt(ButtonNo).id != c.ToString() + r.ToString())
                                 {
                                     if (BlankButtonCount < 2)
```
`if (BlankButtonCount == 1 )` does make more sense so I've changed it to that.

zelmon64 commented on src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs:
```diff
@@ -124,12 +126,13 @@ private static void CKPageFileChanged(DependencyObject dependencyObject, Depende
                                         Log.DebugFormat("Blank button {3} added at column {0} row {1} with background colour {2}.", c, r, blankbutton.background_color, BlankButtonCount);
                                     }
                                     blankbutton.id = c.ToString() + r.ToString();
```
The inserted buttons are now individually instantiated and the properties are immediately set. The `Buttons` class now also contains defined defaults.

zelmon64 commented:
@kmcnaught thanks for your comments/suggestions. I agree with all of them and have modified the code accordingly. I also simplified where the blank keys are inserted and added default values to the `Buttons` class.

I would love your thoughts on these changes.

kmcnaught commented:
@zelmon64 Great, thanks, I'm happy with everything now except for the spelling of the new variable `CurentButtonIsBlank` :)
